components/authz-service/server/v2/project.go

```diff
@@ -331,7 +331,16 @@ func (s *ProjectState) ListProjectsForIntrospection(
 func (s *ProjectState) DeleteProject(ctx context.Context,
 	req *api.DeleteProjectReq) (*api.DeleteProjectResp, error) {
 
-	err := s.ProjectPurger.Start(req.Id)
+	resp, err := s.ListRulesForProject(ctx, &api.ListRulesForProjectReq{Id: req.Id})
```

Comment:
This is the main change. Checking if there are any applied or staged rules for the project to be deleted.